lib/analyze-scope.js

```diff
@@ -323,7 +323,7 @@ module.exports = function(ast, parserOptions) {
         parserOptions.ecmaFeatures.globalReturn) === true,
     impliedStrict: false,
     sourceType: ast.sourceType,
-    ecmaVersion: parserOptions.ecmaVersion || 2018,
+    ecmaVersion: parserOptions.ecmaVersion,
```

Comment:
This is already set by https://github.com/babel/babel-eslint/pull/743/files#diff-6d186b954a58d5bb740f73d84fe39073R29